### PR TITLE
Add certs to the response construct tx endpoint

### LINKED ISSUE #4871: The constructTransaction endpoint doesn't contain certificates in response
### Just checking...

- [X] This is a `cardano-wallet` bug.
- [X] I am using the latest `cardano-wallet` release.
- [X] I am using the *correct* `cardano-node` version for that release of `cardano-wallet`.

### Version

v2024-11-18

### Platform

macOS 14

### Installation method

Nix

### Network configuration

preview

### Context

_No response_

## Steps to Reproduce

Call the [constructTransaction](https://cardano-foundation.github.io/cardano-wallet/api/edge/#operation/constructTransaction) endpoint with `{ vote: 'abstain' }` payload

### Expected behavior

The response contains a `coin_selection.certificates` with appropriate certificates.

### Actual behavior

 - no `coin_selection.certificates` property.
 - the cbor received under `transaction` property has the proper certificates.

### PR DESCRIPTION
**Pull request by paweljakubas**
- [x] Adding voting certs to the response of constructTransaction (until now only delegation certs were supported)
- [x] test all cases in the existent integration tests

### Issue Number
fix https://github.com/cardano-foundation/cardano-wallet/issues/4871

### CODE REVIEW

File: lib/api/src/Cardano/Wallet/Api/Http/Shelley/Server.hs [hunk 1]
```diff
@@ -4593,21 +4595,39 @@ mkApiCoinSelection deps refunds mcerts metadata unsignedTx =
...
             <$> refunds
         , metadata = ApiBytesT. serialiseToCBOR
             <$> metadata
         }
-
-mkApiCoinSelectionCerts
+  where
+    delCertsM = uncurry mkApiCoinSelectionDelCerts <$> mDelCerts
+    votingCertsM = uncurry mkApiCoinSelectionVotingCerts <$> mVotingCerts
+    mergeCerts del vote = case (del, vote) of
```
Comment (abailly) [inline lib/api/src/Cardano/Wallet/Api/Http/Shelley/Server.hs:4610]: I would really want to have tests for this :)

Comment (abailly) [reply lib/api/src/Cardano/Wallet/Api/Http/Shelley/Server.hs:4610]: And I mean unit test. I imagine the various integration voting tests are there to cover various cases, but it woiuld be much more efficient and clear to cover those cases in a unit test dedicated to the merge function, and have only _one_ integration test that ensures this endpoint is hit

**Review by abailly — COMMENTED** [review]
Suggested covering the various merge cases with unit test rather than integration tests

File: lib/api/src/Cardano/Wallet/Api/Http/Shelley/Server.hs (hunk 1 above)
Comment (paweljakubas) [reply lib/api/src/Cardano/Wallet/Api/Http/Shelley/Server.hs:4610]: in the end, I have come to conclusion that we should present deleg/voting certs as they are (this is constructTransaction), plus we want the same result as in the case of decodeTransaction. So `mergeCerts` is just combined via `semigroup <>`. So no logic really.

File: lib/integration/scenarios/Test/Integration/Scenario/API/Voting.hs [hunk 2]
```diff
@@ -227,10 +227,16 @@ spec = describe "VOTING_TRANSACTIONS" $ do
             }|]
         rTx1 <- request @(ApiConstructTransaction n) ctx
             (Link.createUnsignedTransaction @'Shelley src) Default delegationJoin
+
```
Comment (abailly) [inline lib/integration/scenarios/Test/Integration/Scenario/API/Voting.hs:230]: This test is 250 lines long, it's pretty hard to follow and understand what's going exactly. It seems the change introduced to do not change the logic, only the structure of the results?

**Review by abailly — APPROVED** [review]
